src/core/server.py

```
@@ -153,63 +177,105 @@ async def crawl4ai_lifespan(server: FastMCP) -> AsyncIterator[Crawl4AIContext]:
                 
                 logger.info("✓ Knowledge graph components initialized")
             except Exception as e:
-                logger.error(f"Failed to initialize Neo4j components: {format_neo4j_error(e)}")
+                logger.error("Failed to initialize Neo4j components: %s", format_neo4j_error(e))
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)